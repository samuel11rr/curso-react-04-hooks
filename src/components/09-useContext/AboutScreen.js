import React from 'react'
import { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

  const {user, setuser} = useContext(UserContext);
  const handleClick = () => {
    setuser({});
  }

  return (
    <div>
      <h1> About Screen </h1>
      <hr/>
      
      <pre>
        { JSON.stringify(user, null, 3) }
      </pre>

      <button 
        className="btn btn-outline-danger"
        onClick={ handleClick }
      >
        Logout
      </button>
    </div>
  )
}
