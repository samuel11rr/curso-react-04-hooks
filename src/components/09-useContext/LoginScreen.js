import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const { user, setuser } = useContext(UserContext)
  console.log(user);

  return (
    <div>
      <h1> Login Screen </h1>
      <hr/>
      
      <button 
        className="btn btn-primary" 
        onClick={ () => setuser({
          id: 123,
          name: 'Sam',
          email: 'sam@algo.com'
        }) }
      >
        Login
      </button>
    </div>
  )
}
