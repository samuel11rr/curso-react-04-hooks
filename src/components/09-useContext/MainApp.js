import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

  const [user, setuser] = useState({})

  return (
    <UserContext.Provider value={{
      user,
      setuser
    }}>
      <AppRouter />
    </UserContext.Provider>
  )
}
