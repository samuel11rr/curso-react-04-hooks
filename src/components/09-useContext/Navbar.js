import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">useContex</Link>

      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item">
            <NavLink className="nav-link" activeClassName="active" to="/" exact>Home</NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" activeClassName="active" to="/about" exact>About</NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" activeClassName="active" to="/login" exact>Login</NavLink>
          </li>
        </ul>
      </div>
    </nav>
  )
}
