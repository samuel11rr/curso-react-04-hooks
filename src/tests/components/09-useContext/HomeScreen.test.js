import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {
  const user = {
    name: 'Samuel',
    email: 'samuel@mail.com'
  }

  const wrapper = mount(
    <UserContext.Provider
      value={{ user }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );

  test('debe mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  
})
