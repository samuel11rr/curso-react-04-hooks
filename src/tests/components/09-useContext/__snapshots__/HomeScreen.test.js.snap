// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
     Home Screen 
  </h1>
  <hr />
  <pre>
    {
   "name": "Samuel",
   "email": "samuel@mail.com"
}
  </pre>
</div>
`;
